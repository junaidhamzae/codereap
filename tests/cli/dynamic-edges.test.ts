import path from 'node:path';
import fs from 'node:fs';
import os from 'node:os';
import { runCli } from '../helpers/cli';

function getRow(json: any[], endsWith: string) {
  return (json as any[]).find((r) => r.node.endsWith(endsWith));
}

describe('CLI dynamic edges on/off', () => {
  it('marks C.ts live by default and orphan when dynamicEdges off', () => {
    const fixture = path.resolve(__dirname, '../fixtures/dynamic-literals');
    const tmp = fs.mkdtempSync(path.join(os.tmpdir(), 'cli-'));
    const outBase = path.join(tmp, 'report');

    // default: dynamic edges ON
    let res = runCli(['--root', fixture, '--out', outBase]);
    expect(res.status).toBe(0);
    let rows = res.json as any[];
    expect(getRow(rows, 'C.ts').orphan).toBe(false);

    // with dynamicEdges off
    res = runCli(['--root', fixture, '--out', outBase, '--dynamicEdges', 'off']);
    expect(res.status).toBe(0);
    rows = res.json as any[];
    expect(getRow(rows, 'C.ts').orphan).toBe(true);
  });
});


