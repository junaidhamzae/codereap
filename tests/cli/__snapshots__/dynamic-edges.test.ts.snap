// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`CLI dynamic edges on/off marks C.ts live by default and orphan when dynamicEdges off 1`] = `
[
  {
    "exists": true,
    "in-degree": 0,
    "node": "A.ts",
    "orphan": false,
    "size-bytes": 59,
    "symbols": {
      "exports": {
        "default": {
          "exists": false,
          "orphan": true,
          "referencedInFile": false,
        },
        "named": [
          {
            "name": "load",
            "orphan": true,
            "reexport": false,
            "referencedInFile": false,
          },
        ],
        "reExports": [],
      },
    },
  },
  {
    "exists": true,
    "in-degree": 0,
    "node": "C.ts",
    "orphan": true,
    "size-bytes": 27,
    "symbols": {
      "exports": {
        "default": {
          "exists": false,
          "orphan": true,
          "referencedInFile": false,
        },
        "named": [
          {
            "name": "dyn",
            "orphan": true,
            "reexport": false,
            "referencedInFile": false,
          },
        ],
        "reExports": [],
      },
      "imports": [],
    },
  },
  {
    "exists": true,
    "in-degree": 0,
    "node": "package.json",
    "orphan": true,
    "size-bytes": 110,
    "symbols": {
      "exports": {
        "default": {
          "exists": false,
          "orphan": true,
          "referencedInFile": false,
        },
        "named": [],
        "reExports": [],
      },
      "imports": [],
    },
  },
]
`;
